api/v1beta1/azuremanagedcluster_types.go

```diff
@@ -37,6 +37,12 @@ type AzureManagedClusterStatus struct {
 }
 
 // +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="Cluster",type="string",JSONPath=".metadata.labels.cluster\\.x-k8s\\.io/cluster-name",description="Cluster to which this AzureManagedCluster belongs"
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
```

Comment:
This doesn't seem to work on an `AzureManagedCluster` because its status field doesn't apparently have `conditions`:

```shell
k get azuremanagedcluster,azuremanagedcontrolplane
NAME                                                            CLUSTER     READY   SEVERITY   REASON   AGE
azuremanagedcluster.infrastructure.cluster.x-k8s.io/aks-17271   aks-17271                               13m

NAME                                                                 CLUSTER     READY   SEVERITY   REASON   AGE
azuremanagedcontrolplane.infrastructure.cluster.x-k8s.io/aks-17271   aks-17271   True                        13m

% k get azuremanagedcluster,azuremanagedcontrolplane -o yaml
apiVersion: v1
items:
- apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
  kind: AzureManagedCluster
  <...>
  status:
    ready: true
- apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
  kind: AzureManagedControlPlane
  <...>
  status:
    conditions:
    - lastTransitionTime: "2024-01-24T00:00:19Z"
      status: "True"
      type: Ready
    - lastTransitionTime: "2024-01-24T00:00:19Z"
      status: "True"
      type: ManagedClusterRunning
    - lastTransitionTime: "2024-01-23T23:52:27Z"
      status: "True"
      type: ResourceGroupReady
    - lastTransitionTime: "2024-01-23T23:53:28Z"
      status: "True"
      type: SubnetsReady
    - lastTransitionTime: "2024-01-23T23:52:58Z"
      status: "True"
      type: VNetReady
    initialized: true
    ready: true
    version: v1.28.3
```

Reply:
:/

Reply:
I apologize I didn't notice that so should I remove all fields like 'Ready', 'Severity', and 'Reason'?

Reply:
> so should I remove all fields ?

I think we have to, since conditions aren't supported here. We can keep "Ready" as just a boolean field though.